### PR TITLE
feat(RHINENG-13864): Load activation keys

### PR DESCRIPTION
**Pull request by johnsonm325**
This PR adds the ability to load activation keys from the Activation Keys UI app and allows a user to select one of those activation keys to get a tailored registration assistant experience.

When an activation key is selected, a dropdown should appear below the select box giving details on the key. And stepping through the Progress Stepper should eventually show the activation key name in the Code Block for the commands.

A user may also auto-generate a key from the dropdown menu. This will create a key with the name `activation-key-<timestamp>` which will be automatically selected for the user.

A detailed activation key can also be created via the dropdown. The create activation key modal will be opened and a key can be detailed and created. After the modal is closed, the newly created key will be automatically selected for the user.

### CODE REVIEW

File: src/Components/RegistrationProgress/FirstStep/FirstStep.js [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+  const handleActivationKeySelect = (selectedKeyDetails) => {
+    setSelectedKey(selectedKeyDetails);
+    step === 0 && setStep(1);
+    setOpen(false);
+  };
+
+  useEffect(() => {
+    if (createdKeyName) {
+      keys.map((key) => {
```
Comment (bastilian) [inline src/Components/RegistrationProgress/FirstStep/FirstStep.js:62]: A for loop would be more appropriate here.

File: src/App.js [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
     insights.chrome.hideGlobalFilter?.();
   }, []);
 
-  return <RouteList childProps={props} />;
+  useEffect(() => {
+    if (chrome) {
+      const registry = getRegistry();
+      registry.register({ notification: notificationsReducer });
+    }
+  }, [chrome]);
```
Comment (bastilian) [inline src/App.js:21]: Why is this required? The store `init` function should already have the notificationsReducer initialised, does it not?

File: src/Components/RegistrationProgress/RegAssistantFooter.js [hunk 3]
```diff
@@ -1,26 +1,30 @@
...
-        >
-          Remote Host Configuration and Management | Red Hat Product
-          Documentation{' '}
-          <Icon className="pf-v5-u-ml-xs">
-            <ExternalLinkAltIcon />
-          </Icon>
-        </a>
-      </Text>
+      {(operatingSystem.id === 'centos-radio' ||
+        operatingSystem.id === 'rhel9-radio') && (
```
Comment (bastilian) [inline src/Components/RegistrationProgress/RegAssistantFooter.js:11]: If we can not avoid hardcoding this, we should at least extract these strings and put them into a descriptive constant.

**Comment by johnsonm325** [comment]
@bastilian changes pushed

**Comment by bastilian** [comment]
@johnsonm325 Would you mind adding step by step instructions on how to test this? I've looked at the `/insights/connector/activation-keys` route, but nothing changed there. Where can I find the place where users are able to select activation keys?

**Review by bastilian — APPROVED** [review]
LGTM! Thank you @johnsonm325!
